fornax-hea/build-sas.sh

```diff
@@ -0,0 +1,182 @@
+
+
+############ Moving unpacked SAS and installing ###########
+# Moves all of the files unpacked from the SAS download into the conda environment directory
+#  for SAS - it is easier to install SAS in-situ, rather than installing it then moving it, as 
+#  some file paths get baked in during the installation process
+mv * $ENV_DIR/sas/
+# We must follow the unpacked files
+cd $ENV_DIR/sas/
+
+# Run the SAS install script, specifically in the environment we've just created
+micromamba run -n sas ./install.sh
+###########################################################
+
+
+# TODO - finalize handling of calibration files. Not just XMM, but all missions. Current version of XMM CCF is
+#  stored in 'support-data', which will do for now
+##################### Download XMM CCF ####################
+# We download it straight into the support data directory - this command cannot work on Fornax images without
+#  ensuring we install rsync through Conda, which is why we added it to the sas conda env
+# The ';' on the end of the rsync call means that the rest of the script will proceed if this command fails
+#mkdir -p $SUPPORT_DATA_DIR/xmm_sas/
+#micromamba run -n sas rsync -v -a --delete --delete-after --force --include='*.CCF' --exclude='*/' sasdev-xmm.esac.esa.int::XMM_VALID_CCF $SAS_CCFPATH;
+
+# Rather than the rsync method, we'll download all .CCF files from the HEASARC mirror
+#  of XMM calibration files using wget. This doesn't necessarily seem as safe
+#  as the rsync method, but will do for. This command is set up to recursively download level-1 (i.e. not going 
+#  down into sub-directories) files that have the .CCF extension. Sub-directories are not downloaded, and the files
+#  are put into the $SAS_CCFPATH path set at the top of this script.
+# wget -r -l1 -nd --accept .CCF -P $SAS_CCFPATH -e robots=off https://heasarc.gsfc.nasa.gov/FTP/caldb/data/xmm/ccf/
+###########################################################
+
+
+################ Add conda (de)activation scripts ###############
+# These scripts set up SAS and handles additional environment variable setting
+cat <<EOF > $ENV_DIR/sas/etc/conda/activate.d/sas-general_activate.sh
+#!/usr/bin/bash
+
+# SAS can be very particular about Perl - this is the path we set when SAS was 'built'
+export SAS_PERL=/usr/bin/perl
+# And this is the conda environment we set up for it
+export SAS_PYTHON=\$ENV_DIR/sas/bin/python
+
+# Adds the SAS conda environment library to the library path - without this
+#  we will get errors about not being able to find libsm.so.6 
+export LD_LIBRARY_PATH="\$LD_LIBRARY_PATH:$ENV_DIR/sas/lib"
+
+# Any attempted init of SAS will fail without this path being set
+export SAS_DIR=\$ENV_DIR/sas/${sas_install_dir}
+source \$SAS_DIR/setsas.sh
+EOF
+
+
+cat <<EOF > $ENV_DIR/sas/etc/conda/activate.d/sas-ccf_activate.sh
+#!/usr/bin/bash
+
+# This sets the environment variable for the XMM Current Calibration Files (CCF)
+export SAS_CCFPATH=${SAS_CCFPATH}
```

Comment:
Please reference `SUPPORT_DATA_DIR ` explicitly (escaped $) so that this is not broken if its value changes in the image.